datastores-parent/datastore-hazelcast/src/main/java/org/wicketstuff/datastores/hazelcast/HazelcastDataStore.java

```diff
@@ -41,54 +57,78 @@
 	 *
 	 * @param hazelcast     The hazelcast instance
 	 */
-	public HazelcastDataStore(HazelcastInstance hazelcast)
+	public HazelcastDataStore(String applicationName, HazelcastInstance hazelcast)
 	{
+		super(applicationName);
+		
 		this.hazelcast = Args.notNull(hazelcast, "hazelcast");
+		
+		SerializerConfig config = new SerializerConfig().
+		        setImplementation(new SerializedPageSerializer()).
+		        setTypeClass(SerializedPage.class);
+		hazelcast.getConfig().getSerializationConfig().addSerializerConfig(config);
 	}
 
+	/**
+	 * Pages are always serialized, so versioning is supported.
+	 */
 	@Override
-	public byte[] getData(String sessionId, int pageId)
+	public boolean supportsVersioning()
 	{
-		byte[] bytes = null;
-		IMap<Integer, byte[]> map = hazelcast.getMap(sessionId);
-		if (map != null) {
-			bytes = map.get(pageId);
-		}
+		return true;
+	}
 
-		if (LOGGER.isDebugEnabled()) {
-			LOGGER.debug("Got {} for session '{}' and page id '{}'",
-				new Object[] {bytes != null ? "data" : "'null'", sessionId, pageId});
+	@Override
+	protected IManageablePage getPersistedPage(String sessionIdentifier, int id) {
+		IMap<Integer, SerializedPage> map = hazelcast.getMap(sessionIdentifier);
+		if (map != null)
+		{
+			SerializedPage page = map.get(id);
+			
+			if (LOGGER.isDebugEnabled())
```

Comment:
I believe checks like this should be used only if there is loop inside or something heavy